File: calculator.py
def calculator():
    a = float(input("enter the first number:"))
    b = float(input("enter the second number:"))
    print("1. addtion[+] , 2. subtraction[-] , 3. multiplication[*] , 4. division[/]")
    c = int(input("enter the desired number for operator"))
    match c:
        case 1:
            print("the addition is:", a + b)
        case 2:
            print("the subtraction is:", a - b)
        case 3:
            print("the multiplication is:", a * b)
        case 4:
           try:
               print("the division is:", a / b)
           except ZeroDivisionError:
               print("Error: Division by zero is not allowed.")
        case _:
            print("invalid operator")

calculator()

File: calculator.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	char ch;
	double a, b;
	while (1) {
		printf("Enter an operator (+, -, *, /) \n");
		scanf(" %c", &ch);

		
		printf("Enter two first and second operand: ");
		scanf("%lf %lf", &a, &b);

		switch (ch) {

		
		case '+':
			printf("%.1lf + %.1lf = %.1lf\n", a, b, a + b);
			break;

		case '-':
			printf("%.1lf - %.1lf = %.1lf\n", a, b, a - b);
			break;

		case '*':
			printf("%.1lf * %.1lf = %.1lf\n", a, b, a * b);
			break;

		case '/':
			printf("%.1lf / %.1lf = %.1lf\n", a, b, a / b);
			break;

		default:
			printf(
				"Error! please write a valid operator\n");
		}

		printf("\n");
	}
}
